app/lib/support_ticket.rb

```diff
@@ -0,0 +1,58 @@
+require "gds_zendesk/client"
+require "gds_zendesk/dummy_client"
+
+class SupportTicket
+  def self.client
+    @client ||= if Rails.env.development?
+                  GDSZendesk::DummyClient.new(logger: Rails.logger)
+                else
+                  credentials = ZENDESK_CREDENTIALS.merge(logger: Rails.logger)
+                  GDSZendesk::Client.new credentials
+                end
+  end
+
+  def self.send(hash)
+    new(hash).send
+  end
+
+  attr_reader :body
+
+  def initialize(hash)
+    @body = construct_body(**hash)
+  end
+
+  def payload
+    {
+      subject: "Named contact",
+      tags: %w[public_form named_contact],
+      priority: "normal",
+      comment: { body: },
+    }
+  end
+
+  def construct_body(requester:, details:, link:, javascript_enabled:, referrer: "Unknown", user_agent: "Unknown", **)
+    <<~MULTILINE_STRING
+      [Requester]
+      #{"#{requester[:name]} <#{requester[:email]}>"}
+
+      [Details]
+      #{details}
+
+      [Link]
+      #{link}
+
+      [Referrer]
+      #{referrer}
+
+      [User agent]
+      #{user_agent}
+
+      [JavaScript Enabled]
+      #{javascript_enabled}
+    MULTILINE_STRING
+  end
+
+  def send
+    self.class.client.zendesk_client.tickets.create!(payload)
```

Comment:
This looks a bit intense - ties into my previous comment about the static client